### ISSUE TITLE
Potential bug for continuous generation

### ISSUE THREAD

**Issue by hojjatkarami** [btheodorou99/HALO_Inpatient#15]
Hi,

Congrats on the very replicable code!
I just have a concern regarding checking for the end token generation:

https://github.com/btheodorou99/HALO_Inpatient/blob/0886b296a5ad8882552edd2f4b017dfde94856f1/continuous_variables/generate.py#L35

Apparently, this checks if the end visit is generated for all samples in the batch. While this is correct for Without-lab case:
https://github.com/btheodorou99/HALO_Inpatient/blob/0886b296a5ad8882552edd2f4b017dfde94856f1/test_model.py#L177

,I think  for with-lab case, it should be:
```python
 if torch.sum(torch.sum(prev[:,:config.code_vocab_size+ config.lab_vocab_size+ config.continuous_vocab_size+ config.label_vocab_size+ 1], dim=1).bool().int(), dim=0).item() == batch_size: 
```

**Comment by btheodorou99** [comment]
Thank you very much for catching this! I have updated the code with the fix. In practice, this early stopping doesn't hit much anyways (including labs means that someone in the batch will generally run out of context), but your check for the with-lab case is definitely the right one.

**Comment by hojjatkarami** [comment]
Thank you very much for updating the code.

I have another concern regarding dataset building process:

https://github.com/btheodorou99/HALO_Inpatient/blob/53e643d5128244507ab15557527501f5851d4a09/build_dataset.py#L30

it seems that by using `list(set(***))`, the order of codes in a visit is lost. Here is an example:
![image](https://github.com/btheodorou99/HALO_Inpatient/assets/28216830/276372a3-7e05-47b0-9be6-fd8b85184751)

However, I was thinking that the order is important especially in the code-level module.

**Comment by btheodorou99** [comment]
You're right that the order is important for the architecture in the code-level module. However, this does not sacrifice that as each code is randomly assigned an index (just below that code in lines 40-46) and even if the indices are stored out of order they will fill the matrix with ones in the right order which will maintain the code ordering with respect to the architecture. In terms of the output patient data, the order of the codes in a visit does not matter (semantically they all occur simultaneously and just the visits themselves are sequentially ordered).
